### DISCUSSION TITLE
Simple multi-tenant/VirtualHost question

### DISCUSSION THREAD

**Discussion by oset-bschuhma** [OpenIDC/mod_auth_openidc#664] [Q&A]
Hi there,

I'm new to the module - I'm actually researching it for a customer who uses it.  I've read as much as I can find, but I don't see an answer to this.  Can mod_auth_openidc handle different configurations per VirtualHost?  I think the answer is "Yes", but I wanted to hear it from you Pros before I suggested it to the customer.

I found a simple httpd config in the docs at [https://github.com/zmartzone/mod_auth_openidc/wiki#16-what-does-a-minimal-apache-configuration-file-look-like](url)

Since the OIDC configuration and \<Location\> directives appear within the VirtualHost it appears I could have different configurations for each VirtualHost,  correct?

Customer is trying to protect 50 SPAs, FWIW.  I figured they could add a bunch of VirtualHosts (spa1.xyz.com, spa2.xyz.com, etc.) and configure their OIDC connections per VirtualHost.

Thanks!

Bret

**Comment by oset-bschuhma** [comment]
Gentle nudge... was hoping Hans (@zandbelt) would drop a quick "yes" or "no" in here... Anyone have experience using different configs with multiple VirtualHosts?

Thanks!

Bret

**Comment by zandbelt** [comment]
one can have multiple virtualhosts running in a single Apache instance, each with their own mod_auth_openidc OIDC RP config, one and only one per virtualhost

**Reply by oset-bschuhma** [reply]
Thanks a million, Hans!  The customer said they tried this last summer and the last virtual host config overwrote all the others - they could only have one config, they said.  Sounds like you've made things a lot better over the last year!  :-)

**Comment by zandbelt** [comment]
per-virtual host config has been the default since the first release in 2014...

**Comment by oset-bschuhma** [comment]
Oh, *WOW*... hmmm... I wonder what happened with the customer's config,
then?

Let me be a little more precise...  If a customer has a SPA served from
/www/spa1, and that's set up as a virtual host as, say, spa1.domain.com and
it's protected with your oidc module, they can also set up another virtual
host for spa2.domain.com with a separate oidc module configuration served
from /www/spa2?  I think I'm restating the obvious, but I just want to make
sure.  Sorry if you feel like this is a stupid question.  :-(

Regards,

Bret
--
Bret Schuhmacher | Sr. Member of Technical Staff | OSET Institute | Mobile:
865.257.9856 | Main: 650.600.1450
